dotnet/samples/Concepts/Filtering/TelemetryWithFilters.cs

```diff
@@ -0,0 +1,153 @@
+/// </summary>
+public class TelemetryWithFilters(ITestOutputHelper output) : BaseTest(output)
+{
+    [Fact]
+    public async Task LoggingAsync()
+    {
+        // Initialize kernel with chat completion service.
+        var builder = Kernel
+            .CreateBuilder()
+            .AddOpenAIChatCompletion("gpt-4", TestConfiguration.OpenAI.ApiKey);
+
+        // Create and add logger, which will output messages to test detail summary window.
+        var logger = this.LoggerFactory.CreateLogger<TelemetryWithFilters>();
+        builder.Services.AddSingleton<ILogger>(logger);
+
+        // Add filters with logging.
+        builder.Services.AddSingleton<IFunctionInvocationFilter, FunctionInvocationLoggingFilter>();
+        builder.Services.AddSingleton<IPromptRenderFilter, PromptRenderLoggingFilter>();
+        builder.Services.AddSingleton<IAutoFunctionInvocationFilter, AutoFunctionInvocationLoggingFilter>();
+
+        var kernel = builder.Build();
+
+        // Import sample functions.
+        kernel.ImportPluginFromFunctions("HelperFunctions",
+        [
+            kernel.CreateFunctionFromMethod(() => DateTime.UtcNow.ToString("R"), "GetCurrentUtcTime", "Retrieves the current time in UTC."),
+            kernel.CreateFunctionFromMethod((string cityName) =>
+                cityName switch
+                {
+                    "Boston" => "61 and rainy",
+                    "London" => "55 and cloudy",
+                    "Miami" => "80 and sunny",
+                    "Paris" => "60 and rainy",
+                    "Tokyo" => "50 and sunny",
+                    "Sydney" => "75 and sunny",
+                    "Tel Aviv" => "80 and sunny",
+                    _ => "31 and snowing",
+                }, "GetWeatherForCity", "Gets the current weather for the specified city"),
+        ]);
+
+        // Enable automatic function calling.
+        var executionSettings = new OpenAIPromptExecutionSettings { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions };
+
+        // Invoke prompt with arguments.
+        const string Prompt = "Given the current time of day and weather, what is the likely color of the sky in {{$city}}?";
+        var result = await kernel.InvokePromptAsync(Prompt, new(executionSettings) { ["city"] = "Boston" });
+
+        Console.WriteLine(result);
+    }
+
+    /// <summary>
+    /// Filter which logs an information available during function invocation such as:
+    /// Function name, arguments, execution settings, result, duration, token usage.
+    /// </summary>
+    private sealed class FunctionInvocationLoggingFilter(ILogger logger) : IFunctionInvocationFilter
+    {
+        public async Task OnFunctionInvocationAsync(FunctionInvocationContext context, Func<FunctionInvocationContext, Task> next)
+        {
+            long startingTimestamp = Stopwatch.GetTimestamp();
+
+            logger.LogInformation("Function {FunctionName} invoking.", context.Function.Name);
```

Comment:
I wonder how easy it would be for someone to correlate the different messages together here.  Would make sense to use something like a scope here with an id for this function invocation?
https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-scopes

Reply:
Yes, I was thinking about that too, but it's important to understand that scoping is available not only for telemetry in filters, but for out-of-the-box SK telemetry as well, so I wanted to avoid confusion. But since you mentioned it here, I think it makes sense to add this to the example as well. Added scoping with transaction ID.